lms/static/js/ccx/schedule.js

```diff
@@ -251,9 +263,29 @@ var edx = edx || {};
 	    }
 	},
 
+	valid_dates: function(start, due) {
+          var errorList = [];
+          if (!due) {
+            return errorList;
+          } else if (!start && due) {
+            errorList[0] = "Due date cannot be before start date." +
+              " Please enter valid start date and time";
+            return errorList;
+          }
+
+          var dueDate = new Date(due);
+          var startDate = new Date(start);
+
+          if (dueDate < startDate) {
+            errorList[0] = "Due date cannot be before start date.";
+          }
+          return errorList;
+	},
```

Comment:
Question: is this function returning always only one value? If so you do not need to return a list. But I'm not sure about the logic you had in mind.